velox/exec/HashBuild.cpp

```diff
@@ -94,9 +94,13 @@ HashBuild::HashBuild(
   }
 
   // Identify the non-key build side columns and make a decoder for each.
-  const auto numDependents = outputType->size() - numKeys;
-  dependentChannels_.reserve(numDependents);
-  decoders_.reserve(numDependents);
+  const int32_t numDependents = outputType->size() - numKeys;
+  if (numDependents > 0) {
+    // When duplicated keys exist on build side, the size of output type
```

Comment:
The term "duplicated keys" is confusing because it suggests that we are joining on same key multiple types. You have explained that this is not the case, rather we match different columns on probe side with the same column on build side. I wonder if there is a way to make it clearer.

Perhaps,

```
// Number of join keys (numKeys) may be less then number of input columns (inputType->size()). In this case numDependents is negative and cannot be used to call 'reserve'. This happens when we join different probe side keys with the same build side key: SELECT * FROM t LEFT JOIN u ON t.k1 = u.k AND t.k2 = u.k.
```

It would be nice to update PR description as well.

Reply:
Updated.